### PR TITLE
Remove org users who belong to no teams

### LINKED ISSUE #24128: A user should no longer belong to the org after the team he belongs to is deleted
### Description

The user is still a member of org in the org dashboard. But he no longer belongs to any team in the org.

https://user-images.githubusercontent.com/33891828/231987682-cdda51a3-ec35-4fcc-96a8-98469549a502.mov

### Gitea Version

latest

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

build

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by yp05327**
Fix #24128

### CODE REVIEW

File: models/org_team.go [hunk 1]
```diff
@@ -583,16 +575,19 @@ func removeTeamMember(ctx context.Context, team *organization.Team, userID int64
 		}
 	}
 
+	return removeInvaildOrgUser(ctx, e, userID, team.OrgID)
+}
+
+func removeInvaildOrgUser(ctx context.Context, sess db.Engine, userID, orgID int64) error {
```
Comment (lunny) [inline models/org_team.go]: We just need ctx and use `db.GetEngine(ctx)`, so `sess db.Engine` could be removed.`

Comment (yp05327) [reply models/org_team.go]: Already fixed. 😄

File: models/org_team.go [hunk 2]
```diff
@@ -419,6 +419,12 @@ func DeleteTeam(t *organization.Team) error {
 		return err
 	}
 
+	for _, tm := range t.Members {
+		if err := removeInvaildOrgUser(ctx, tm.ID, t.OrgID); err != nil {
```
Comment (silverwind) [inline models/org_team.go]: typo `removeInvalidOrgUser`

File: models/org_team.go [hunk 3]
```diff
@@ -419,6 +419,12 @@ func DeleteTeam(t *organization.Team) error {
 		return err
 	}
 
+	for _, tm := range t.Members {
+		if err := removeInvalidOrgUser(ctx, tm.ID, t.OrgID); err != nil {
```
Comment (wxiaoguang) [inline models/org_team.go:423]: ~~Does it mean that this "team member" doesn't belong to other teams in this org?~~

Hmm ... I see the logic in `removeInvalidOrgUser`, it counts first.

**Review by wxiaoguang — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

**Pull request merged**